### ISSUE TITLE
2.4.0 -> 2.5.0 migration -  log_subscription_filter keeps re-planning "distribution" attribute change

### ISSUE THREAD

**Issue by shorn** [terraform-providers/terraform-provider-aws#8230]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

TF: 0.11.11
AWS provider: 2.5.0 (current infrastructure was stable on version 2.4.0)

### Affected Resource(s)

*  aws_cloudwatch_log_subscription_filter

### Terraform Configuration Files

```hcl
resource "aws_cloudwatch_log_subscription_filter"
"api-svc-ec2-prd-cloudwatch-subscription"
{
  depends_on = [
    "aws_lambda_permission.api-svc-ec2-prd-sumologic-logging-allow-cloudwatch-lambda"
  ]
  name = "api-svc-ec2-prd-cloudwatch-subscription"
  filter_pattern = ""
  log_group_name =
    "${aws_cloudwatch_log_group.api-svc-ec2-prd-loggroup.name}"
  destination_arn =
    "${aws_lambda_function.cloudwatch-sumologic-logging-lambda.arn}"
}
```

### Steps to Reproduce

1. `terraform plan`

Plan output shows that TF is going to set:  `distribution: "ByLogStream" => ""`

2. ` terraform apply`

Apply output shows that TF is setting:  `distribution: "ByLogStream" => ""`

3. `terraform apply`

Plan output shows that TF is going to set:  `distribution: "ByLogStream" => ""`

### Workaround

Update the resource and add attribute `distribution = "ByLogStream"`

**Comment by mbarrien** [comment]
I get this in provider version 2.1.0 too; may be an AWS-side change unrelated to the provider upgrade?

**Comment by tmshn** [comment]
Mmm, I guess this is due to the following change

https://github.com/terraform-providers/terraform-provider-aws/pull/8165/files#diff-f5436f62ee2a2bd5ee984fa092cb808eR163

https://github.com/terraform-providers/terraform-provider-aws/blob/c4d48c0ed67dfa307dca2197ba529fa103bb15b3/aws/resource_aws_cloudwatch_log_subscription_filter.go#L163

Adding `Computed: true` to the argument might help, maybe?

https://github.com/terraform-providers/terraform-provider-aws/blob/c4d48c0ed67dfa307dca2197ba529fa103bb15b3/aws/resource_aws_cloudwatch_log_subscription_filter.go#L54-L57

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.25.0 of the Terraform AWS Provider, in about a week. Thanks to @dedamico for the implementation. 👍
